runtime/interpreter/value.go

```diff
@@ -8093,33 +8093,44 @@ func (v UInt8Value) HashInput(_ *Interpreter, _ func() LocationRange, scratch []
 	return scratch[:2]
 }
 
+func ConvertUnsigned8(
+	memoryGauge common.MemoryGauge,
+	value Value,
+	onOverflow func(),
+	onUnderflow func(),
+) uint8 {
+	switch value := value.(type) {
+	case BigNumberValue:
+		v := value.ToBigInt(memoryGauge)
+		if v.Cmp(sema.UInt8TypeMaxInt) > 0 {
+			onOverflow()
+		} else if v.Sign() < 0 {
+			onUnderflow()
+		}
```

Comment:
Good idea to try to abstract the conversion into a common function. It might be nice to improve it a bit and avoid unnecessary computation and allocations:

Currently the two underflow/overflow functions are always passed. In the Word* case the functions are just empty, which probably causes unnecessary allocation. Also, the overflow and underflow checks are still performed, even though the result is then discarded.

Maybe optimize this and just inline the code, as there's little code that's actually shared, basically just the swith, i.e. have one conversion function for UInt, with the checks, and one for Word, without the checks.